src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services_almost_circular_private.php

```diff
@@ -349,15 +349,13 @@ protected function getManager2Service()
      */
     protected function getManager3Service($lazyLoad = true)
     {
-        $a = ($this->services['listener3'] ?? $this->getListener3Service());
+        $a = new \stdClass();
 
-        if (isset($this->services['manager3'])) {
-            return $this->services['manager3'];
-        }
-        $b = new \stdClass();
-        $b->listener = [0 => $a];
+        $this->services['manager3'] = $instance = new \stdClass($a);
 
-        return $this->services['manager3'] = new \stdClass($b);
+        $a->listener = [0 => ($this->services['listener3'] ?? $this->getListener3Service())];
+
+        return $instance;
```

Comment:
The order of injection in this test case has changed:
manager3 is given a dependency that is missing its listener3 dep.

This is an issue to me, since these test cases are precisely about the order of the steps.

Reply:
I'm not sure to get the issue. Here are the 2 methods. 

```php
    protected function getManager3Service($lazyLoad = true)
    {
        $a = new \stdClass();

        $this->services['manager3'] = $instance = new \stdClass($a);

        $a->listener = [0 => ($this->services['listener3'] ?? $this->getListener3Service())];

        return $instance;
    }

    protected function getListener3Service()
    {
        $this->services['listener3'] = $instance = new \stdClass();

        $instance->manager = ($this->services['manager3'] ?? $this->getManager3Service());

        return $instance;
    }
```

Reply:
Before the change, getManager3 returned `$this->services['manager3'] = new \stdClass($b);`, with $b being already configured (aka `$b->listener` set)
In the new method, the listener propery is set *after* manager3 got its dep.

Reply:
hmmm ok, I get the issue: The object (`$a`) where not fully built at the time it were injected in Manager's constructor.

Reply:
Fixed by https://github.com/symfony/symfony/compare/d077e62d0d56325e67835353b9bd500418705668..de7ff0f3096c7a23d9936f1da4e645f68cf5c586#diff-82d7ebdd6477cecb404b0d3f100a9026e950529edb9a665460ae0330dfad7080R434

And also fix the second change (because I didn't collect ALL circular references for the same node).

Fixtures are identical to previous version. I tested with the projects I used for the 2 previous issues on this.